### ISSUE TITLE
Oil and transmission tempature via Torque Pro

### ISSUE THREAD

**Issue by Step-han** [jilleb/mqb-pm#84]
Hello,

Is it possible to get the option for oil and transmission tempature via Torque Pro? I'm using this app with a BMW F20 m135i and cannot show the oil/transmission now.

Greetings,
Stephan

![preview](https://user-images.githubusercontent.com/54219681/72221550-5060b580-355c-11ea-84ea-632f2c243477.jpg)

**Comment by jilleb** [comment]
Yes, that's possible. I included a couple of Torque transmission and oil temperatures, but I'm not sure if they are compatible with your car. Can you run look up the right PID, by using the Torque Scan plugin?

You will get a list like this, and I need the value between brackets (like [ff124f] )
![image](https://user-images.githubusercontent.com/8352494/72225532-c2e68b00-3586-11ea-8449-8c17266d0a78.png)


Also: nice car! Looks like I need to incorporate a theme like this ;-):
![image](https://user-images.githubusercontent.com/8352494/72225543-ee697580-3586-11ea-9c89-cc541fd661db.png)

**Comment by Step-han** [comment]
I've made same screenshots from my phone with the plugin installed.

Soo it would be [5c] for engine oil temperature and [fe1805] for transmission temperature?

![1](https://user-images.githubusercontent.com/54219681/72225912-165ad800-358b-11ea-952d-a301e86ab7aa.jpg)
![2](https://user-images.githubusercontent.com/54219681/72225913-18249b80-358b-11ea-8993-367372332683.jpg)

Yea the car is amazing, but the infotaiment of BMW sucks, now with a separate module I can have AndroidAuto on the widescreen, now the screen makes sense :)

Theme for BMW would be great!

Thanks for youre quick reply.

**Comment by jilleb** [comment]
Thanks for the screenshots! I'll include these. I'm surprised I didn't include them before, especially since all kinds of exotic PIDs are there (timing advance and such). :-)

ps: dankjewel! :-)

**Comment by Step-han** [comment]
Nice the hear! When u add those PIDs how can I test them? Do you release a new version of the app or can you send me a beta .apk?

**Comment by jilleb** [comment]
I'll send you a beta apk :-)

**Comment by Step-han** [comment]
Sounds promising :)

**Comment by jilleb** [comment]
In the mean time.. I'll draw up some BMW theme ;-) 

![image](https://user-images.githubusercontent.com/8352494/72382792-54d2cd00-371a-11ea-8e69-b27cba2a99bd.png)

inspiration:
![image](https://user-images.githubusercontent.com/8352494/72382855-70d66e80-371a-11ea-806a-a5ea43b9bb75.png)

**Comment by Step-han** [comment]
Looks nice, even for a 15 min playtime in PS.

I saw some commits from yesterday that you adding the new torque PIDS. Does this mean I can test it with a beta .apk? New to github soo don't know how that works.

**Comment by jilleb** [comment]
Almost. I need to do a test drive in my car and then I'll send you a beta.

**Comment by jilleb** [comment]
@Step-han check it out ;-)


[Performance Monitor 0.88a.zip](https://github.com/jilleb/mqb-pm/files/4081383/Performance.Monitor.0.88a.zip)

**Comment by jilleb** [comment]
Hmm, maybe I should try to go for this look:
![image](https://user-images.githubusercontent.com/8352494/72668690-c320d380-3a29-11ea-93f9-96941342b109.png)


What do you think?

**Comment by Step-han** [comment]
I will test it tonight and make some pictures from my original system for inspiration!

**Comment by Step-han** [comment]
I've done a testride. Some points:

- the elements between the clocks don't work anymore
- turbo boost pressure is in bar even if psi is selected. Switching it when the app is on AndroidAuto wil get psi view working but only to the max of 5 psi (my car is doing around 16-17 psi).
- and on the elementsview between the clocks I can't select transmission tempature, only on clocks. Is it possible to select it there?

Here some screens/photos:

![1](https://user-images.githubusercontent.com/54219681/72671655-d9d92180-3a4d-11ea-8f8b-7bf1d9b3871d.jpg)

![2](https://user-images.githubusercontent.com/54219681/72671657-dcd41200-3a4d-11ea-8c62-6241b81df2d1.jpg)

![3](https://user-images.githubusercontent.com/54219681/72671658-de9dd580-3a4d-11ea-8094-38de148fd767.jpg)

My stock clocks from BMW:
![4](https://user-images.githubusercontent.com/54219681/72671660-e0679900-3a4d-11ea-8c9a-b8f82a40f6d0.jpg)

Screenshot of the settings in the app:
![5](https://user-images.githubusercontent.com/54219681/72671666-fecd9480-3a4d-11ea-9bc0-8635721a34d4.jpg)

And thanks for the great work! Oil and transmission temps are working fine in the clock view!

**Comment by jilleb** [comment]
:-) 


yeah, I made some modifications to the torque text-elements and I wasn't 100% sure why I coded them like I did before... so I guess that broke them 🤣 

Thanks for the review! I'll fix them and get back with a new version. 
To be honest, I wish I could scrap the entire unit-support (metric vs emprical) , because it's making the app more complicated than it needs to be 😆

**Comment by Step-han** [comment]
PSI is more accurate then BAR to read off for the turbo boost. Wider range. In the car performance world they always talk in PSI.

Maybe a view of PSI from -15 to +30

I did a drive today with the view of photo 2, coolant, oil, transmission and its nice to see the temperatures. :)

**Comment by Step-han** [comment]
I saw some nice changes like a BMW theme :)

Keep up the good work and many thanks!

**Comment by jilleb** [comment]
:-) yes.. some more test drives tomorrow, and then I'll release. I chose to let Torque decide what unit and min/max range should be used. Hopefully all will work as planned 😁

**Comment by Step-han** [comment]
And how was the testdrive? 😁

**Comment by Step-han** [comment]
Any chance to send me a new beta .apk?

**Comment by Step-han** [comment]
Any news?

**Comment by Step-han** [comment]
I see you added a new public version, I'll will test it today. Will make soms photo's en screenshots for u.

**Comment by Step-han** [comment]
I did a testrun. The Performance Monitor works great!

Only thing is when the performance monitor starts the turbo boost pressure is always in bar even if you have selected PSI. Switch to bar and back to PSI fix it for now. The range of PSI i 0-100 better is -15 to +30 for PSI.

Here some pictures:

![1](https://user-images.githubusercontent.com/54219681/77226861-cc3e3600-6b7b-11ea-8fa9-0128770d4262.jpg)
![2](https://user-images.githubusercontent.com/54219681/77226863-cd6f6300-6b7b-11ea-959a-1a7fd1954a48.jpg)

**Comment by sal989** [comment]
Reviving a old post, any chance this same oil and transmission readings can be done for a 135i n54? So far i have had no luck.
TIA

**Comment by sal989** [comment]
![Uploading Point Blur_Jul132021_113148.jpg…]()
